utils/utils_test.go

```diff
@@ -85,7 +87,8 @@ func TestValidateBrickPathStats(t *testing.T) {
 }
 
 func TestValidateXattrSupport(t *testing.T) {
-	//TODO : xattr related calls need root permission and hence having valid
-	//tests will always fail in build system. Need to find a way to tackle
-	//it.
+	defer tests.Patch(&Setxattr, tests.MockSetxattr).Restore()
+	defer tests.Patch(&Getxattr, tests.MockGetxattr).Restore()
+	defer tests.Patch(&Removexattr, tests.MockRemovexattr).Restore()
+	tests.Assert(t, ValidateXattrSupport("/tmp/b1", "localhost", uuid.NewRandom(), true) == nil)
```

Comment:
Most of the tests are positive tests.  We should also add some negative tests to make sure that the code reacts in the correct way.  For example:

```go
// Do not set, returns nil the first time
var setxattr_err error
defer tests.Patch(&Setxattr, func (path string, attr string, data []byte, flags int) (err error) {
	return setxattr_err
}

// Some tests here

// Now check what happens when setxattr fails
baderror := errors.New("Bad")
setxattr_err = baderror

// Test again
tests.Assert(t, ValidateXattrSupport(...) == baderror)

```